src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -0,0 +1,568 @@
+                _ when attributeValue.Equals("amd64", StringComparison.OrdinalIgnoreCase) => ProcessorArchitecture.Amd64,
+                _ when attributeValue.Equals("arm", StringComparison.OrdinalIgnoreCase) => ProcessorArchitecture.Arm,
+                _ when attributeValue.Equals("msil", StringComparison.OrdinalIgnoreCase) => ProcessorArchitecture.MSIL,
+                _ => ProcessorArchitecture.None
+            };
+            return result != ProcessorArchitecture.None;
+        }
+
+        private static bool TryParseHexNybble(char c, out byte parsed)
+        {
+            int value = HexConverter.FromChar(c);
+            if (value == 0xFF)
+            {
+                parsed = 0;
+                return false;
+            }
+            parsed = (byte)value;
+            return true;
+        }
+
+        private static bool IsWhiteSpace(char ch)
+        {
+            switch (ch)
+            {
+                case '\n':
+                case '\r':
+                case ' ':
+                case '\t':
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool TryGetNextChar(out char ch)
+        {
+            if (_index < _input.Length)
+            {
+                ch = _input[_index++];
+                if (ch == '\0')
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                ch = '\0';
+            }
+
+            return true;
+        }
+
+        //
+        // Return the next token in assembly name. If the result is Token.String,
+        // sets "tokenString" to the tokenized string.
+        //
+        private bool TryGetNextToken(out string tokenString, out Token token)
+        {
+            tokenString = string.Empty;
+            char c;
+
+            while (true)
+            {
+                if (!TryGetNextChar(out c))
+                {
+                    token = default;
+                    return false;
+                }
+
+                switch (c)
+                {
+                    case ',':
+                    {
+                        token = Token.Comma;
+                        return true;
+                    }
+                    case '=':
+                    {
+                        token = Token.Equals;
+                        return true;
+                    }
+                    case '\0':
+                    {
+                        token = Token.End;
+                        return true;
+                    }
+                }
+
+                if (!IsWhiteSpace(c))
+                {
+                    break;
+                }
+            }
+
+#if SYSTEM_PRIVATE_CORELIB
+            ValueStringBuilder sb = new ValueStringBuilder(stackalloc char[64]);
+#else
+            StringBuilder sb = new(64);
```

Comment:
Is 64 a good enough guess to proactively allocate it?

Reply:
Great catch. I've realized that all the projects that reference this file do also reference the `ValueStringBuilder` and removed the usage of `StringBuilder.